prog/vm/nexus.rb

```diff
@@ -437,24 +437,21 @@ def before_run
   end
 
   label def wait_sshable
-    addr = vm.ephemeral_net4
-
-    # Alas, our hosting environment, for now, doesn't support IPv6, so
-    # only check SSH availability when IPv4 is available: a
-    # unistacked IPv6 server will not be checked.
-    #
-    # I considered removing wait_sshable altogether, but (very)
-    # occasionally helps us glean interesting information about boot
-    # problems.
-    hop_create_billing_record unless addr
+    case host.sshable.cmd("common/bin/daemonizer --check wait_sshable_#{q_vm}")
+    when "Succeeded"
+      vm.vm_host.sshable.cmd("common/bin/daemonizer --clean wait_sshable_#{q_vm}")
 
-    begin
-      Socket.tcp(addr.to_s, 22, connect_timeout: 1) {}
-    rescue SystemCallError
-      nap 1
+      hop_create_billing_record
+    when "NotStarted", "Failed"
+      # I considered removing wait_sshable altogether, but (very)
+      # occasionally helps us glean interesting information about boot
+      # problems.
+      prefix_len = vm.ephemeral_net6.netmask.prefix_len + 1
+      random_source_ip = vm.ephemeral_net6.resize(prefix_len).next_sib.nth(SecureRandom.random_number(2**(128 - prefix_len)))
```

Comment:
Interesting, any reason to randomize or should we have the "privileged" prefix be static? It's all ULA anyway (not routed on the Internet). In reality it's something of an internal contract, so it's not high stakes to be somewhat indecisive; it's not an interface with the customer in any meaningful way. But I thought it might be simpler.

Reply:
So, I have opted to use `::3`. I still dynamically add it and remove after the check  is done. The main reason is that I am not really sure the `lo` will always be up whenever we get into `wait_sshable` state. It made sense to always add with `replace` while doing so, I didn't want to leave it hanging in there. It should be ok.

Reply:
Sure, `::3` I guess, it's a pretty changable decision, though I thought you might want to carve out a hextet for "privileged firewall bypass" zone (e.g. `1::2`, whereas we currently use `::2` aka `0::2`).